recipes/repository.rb

```diff
@@ -26,17 +26,25 @@
     action :install
   end
 
-  uri = node['datadog']['agent6'] ? node['datadog']['agent6_aptrepo'] : node['datadog']['aptrepo']
-  distribution = node['datadog']['agent6'] ? node['datadog']['agent6_aptrepo_dist'] : node['datadog']['aptrepo_dist']
-  components = node['datadog']['agent6'] ? ['main', '6'] : ['main']
+  case node['datadog']['agent_major_version'].to_i
+  when 7
+    components = ['7']
+  when 6
+    components = ['main', '6']
```

Comment:
Do we really need `main` when installing Agent 6? I thought it only contained Agent 5 and lower packages.
For example, the ansible module only uses the `6` component to install Agent 6: https://github.com/DataDog/ansible-datadog/blob/dbe9a8fa0e03831285146483404b4520670a2174/defaults/main.yml#L24

Reply:
Apparently old versions of Agent 6 were deployed to main. We should check if all the versions are in `6` before removing it. I'll add a comment.

Reply:
Olivier checked and it can be removed 👌